plugin/trino-pulsar/src/main/java/io/trino/plugin/pulsar/PulsarColumnHandle.java

```diff
@@ -0,0 +1,255 @@
+    }
+
+    @JsonProperty
+    public String getCatalogName()
+    {
+        return catalogName;
+    }
+
+    @Override
+    @JsonProperty
+    public String getName()
+    {
+        return name;
+    }
+
+    @Override
+    @JsonProperty
+    public String getMapping()
+    {
+        return mapping;
+    }
+
+    @Override
+    @JsonProperty
+    public String getDataFormat()
+    {
+        return dataFormat;
+    }
+
+    @Override
+    @JsonProperty
+    public Type getType()
+    {
+        return type;
+    }
+
+    @JsonProperty
+    public boolean isHidden()
+    {
+        return hidden;
+    }
+
+    @Override
+    @JsonProperty
+    public boolean isInternal()
+    {
+        return internal;
+    }
+
+    @Override
+    @JsonProperty
+    public String getFormatHint()
+    {
+        return formatHint;
+    }
+
+    @JsonProperty
+    public HandleKeyValueType getHandleKeyValueType()
+    {
+        return handleKeyValueType;
+    }
+
+    @JsonIgnore
+    public boolean isKey()
+    {
+        return Objects.equals(handleKeyValueType, HandleKeyValueType.KEY);
+    }
+
+    @JsonIgnore
+    public boolean isValue()
+    {
+        return Objects.equals(handleKeyValueType, HandleKeyValueType.VALUE);
+    }
+
+    ColumnMetadata getColumnMetadata()
+    {
+        return new PulsarColumnMetadata(name, type, null, null, hidden,
+                internal, handleKeyValueType, new PulsarColumnMetadata.DecoderExtraInfo(
+                mapping, dataFormat, formatHint));
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        PulsarColumnHandle that = (PulsarColumnHandle) o;
+
+        if (hidden != that.hidden) {
+            return false;
+        }
+        if (internal != that.internal) {
+            return false;
+        }
+        if (catalogName != null ? !catalogName.equals(that.catalogName) : that.catalogName != null) {
+            return false;
+        }
+        if (name != null ? !name.equals(that.name) : that.name != null) {
+            return false;
+        }
+        if (type != null ? !type.equals(that.type) : that.type != null) {
+            return false;
+        }
+        if (mapping != null ? !mapping.equals(that.mapping) : that.mapping != null) {
+            return false;
+        }
+        if (dataFormat != null ? !dataFormat.equals(that.dataFormat) : that.dataFormat != null) {
+            return false;
+        }
+
+        if (formatHint != null ? !formatHint.equals(that.formatHint) : that.formatHint != null) {
+            return false;
+        }
+
+        return Objects.equals(handleKeyValueType, that.handleKeyValueType);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        int result = catalogName != null ? catalogName.hashCode() : 0;
```

Comment:
Objects.hashCode, here and everywhere else

Reply:
Regenerated with IntelliJ, now using Objects.hash or Objects.hashCode